src/cswinrt/strings/WinRT.cs

```diff
@@ -42,34 +54,94 @@ internal sealed class Platform
         internal static extern bool FreeLibrary(IntPtr moduleHandle);
 
         [DllImport("kernel32.dll", EntryPoint = "GetProcAddress", SetLastError = true, BestFitMapping = false)]
-        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, [MarshalAs(UnmanagedType.LPStr)] string functionName);
+        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, sbyte* functionName);
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, ReadOnlySpan<byte> functionName)
+        {
+            fixed (byte* lpFunctionName = functionName)
+            {
+                return TryGetProcAddress(moduleHandle, (sbyte*)lpFunctionName);
+            }
+        }
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, string functionName)
+        {
+            fixed (byte* lpFunctionName = Encoding.UTF8.GetBytes(functionName))
```

Comment:
This is extremely inefficient, and it will allocate and throw away a buffer for the name every time.

Reply:
What do you think would be a better way to do this? From looking at `LibraryImport`, it seems to stack allocate the UTF-8 string if it's small and heap allocate it otherwise.

Reply:
That's a reasonable approach, since most method names would be small enough anyway. Could pick a given threshold (maybe 0x100 like `LibraryImport`) and use that with a `stackalloc` if the name is small enough, or just allocate a new array if not, which would realistically never happen.

Reply:
Does this seem reasonable?
```cs
internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, string functionName)
{
    bool allocated = false;
    Span<byte> buffer = stackalloc byte[0x100];
    if (functionName.Length * 3 >= 0x100) // Maximum of 3 bytes per UTF-8 character, stack allocation limit of 256 bytes (including the null terminator)
    {
        // Calculate accurate byte count when the provided stack-allocated buffer is not sufficient
        int exactByteCount = checked(Encoding.UTF8.GetByteCount(functionName) + 1); // + 1 for null terminator
        if (exactByteCount > 0x100)
        {
#if NET6_0_OR_GREATER
            buffer = new((byte*)NativeMemory.Alloc((nuint)exactByteCount), exactByteCount);
#else
            buffer = (byte*)Marshal.AllocHGlobal(exactByteCount);
#endif
            allocated = true;
        }
    }

    var rawByte = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(buffer));

    int byteCount = Encoding.UTF8.GetBytes(functionName, buffer);
    buffer[byteCount] = 0;

    void* functionPtr = TryGetProcAddress(moduleHandle, (sbyte*)rawByte);

    if (allocated)
#if NET6_0_OR_GREATER
        NativeMemory.Free(rawByte);
#else
        Marshal.FreeHGlobal((IntPtr)rawByte);
#endif

    return functionPtr;
}
```

Reply:
This looks reasonable but might want to confirm that the version of `Encoding.UTF8.GetBytes` you are using is accessible in NET Standard 2.0.  It is fine if the .NET Standard version is not blittable / AOT complaint if there is no equivalent.

Reply:
I think what I'll do is pin the string on .NET Standard 2.0 and use the `GetBytes(Char*, Int32, Byte*, Int32)` function.